File: Code/Components/DanpheEMR.ServerModel/ClinicalModels/ClinicalEye/RefractionModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class RefractionModel
    {
        [Key]
        public int Id { get; set; }
        public int MasterId { get; set; }
        public DateTime? Date { get; set; }
        public string TimePoint { get; set; }
        public int? UCVA { get; set; }
        public string ULett { get; set; }
        public string NUC { get; set; }
        public double? Sph { get; set; }
        public double? Cyf { get; set; }
        public int? Axis { get; set; }
        public int? BSCVA { get; set; }
        public string BLett { get; set; }
        public string DCNV { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public Boolean IsOD { get; set; }

    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical/eye-examination/shared/Refraction.model.ts
import {
  NgForm,
  FormGroup,
  FormControl,
  Validators,
  FormBuilder,
  ReactiveFormsModule
} from '@angular/forms';
import * as moment from 'moment/moment';

export class RefractionModel {
  public Id: number = 0;
  public MasterId: number = 0;
  public Date: string;
  public TimePoint: string = null;
  public UCVA: number = 0;
  public ULett: string = null;
  public NUC: string = null;
  public Sph: number = 0;
  public Cyf: number = 0;
  public Axis: number = 0;
  public BSCVA: number = 0;
  public BLett: string = null;
  public DCNV: string = null;
  public CreatedBy: number = 0;
  public CreatedOn: Date = new Date();
  public IsOD: boolean;

  constructor() {

  }
  dateValidator(control: FormControl): { [key: string]: boolean } {
    var currDate = moment().format('YYYY-MM-DD HH:mm');
    if (control.value) { // gets empty string for invalid date such as 30th Feb or 31st Nov)
      if ((moment(control.value).diff(currDate) > 0)
        || (moment(currDate).diff(control.value, 'years') > 200)) //can select date upto 200 year past from today.
        return { 'wrongDate': true };
    }
    else
      return { 'wrongDate': true };
  }

}
